internal/ingress/controller/kong.go

```diff
@@ -1265,18 +1265,27 @@ func deleteServiceUpstream(host string, client *kong.Client) error {
 // the persisted state in the Kong database
 // This is required because a plugin has defaults that could not exists in the CRD.
 func pluginDeepEqual(config map[string]interface{}, kong *kong.Plugin) bool {
-	for k, v := range config {
+	configToCompare := make(map[string]interface{})
+	for k, _ := range config {
 		kv, ok := kong.Config[k]
 		if !ok {
 			return false
 		}
+		// add kv pairs which config and kong.Config both have
+		// to emit any default keys
+		configToCompare[k] = kv
```

Comment:
Question: The `config` properties of Plugins in Kong can be nested. Will this take care of those cases?